client/client.go

```diff
@@ -0,0 +1,137 @@
+// Client is a cross-platform client for the signer binary (a.k.a."EnterpriseCertSigner").
+// The signer binary is OS-specific, but exposes a standard set of APIs for the client to use.
+package client
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"net/rpc"
+	"os"
+	"os/exec"
+
+	"github.com/googleapis/enterprise-certificate-proxy/client/util"
+)
+
+const signAPI = "EnterpriseCertSigner.Sign"
+const certificateChainAPI = "EnterpriseCertSigner.CertificateChain"
+const publicKeyAPI = "EnterpriseCertSigner.Public"
+
+// A Transport wraps a pair of unidirectional streams as an io.ReadWriteCloser.
+type Transport struct {
+	io.ReadCloser
+	io.WriteCloser
+}
+
+// Close closes t's underlying ReadCloser and WriteCloser.
+func (t *Transport) Close() error {
+	rerr := t.ReadCloser.Close()
+	werr := t.WriteCloser.Close()
+	if rerr != nil {
+		return rerr
+	}
+	return werr
+}
+
+func init() {
+	gob.Register(crypto.SHA256)
+	gob.Register(&rsa.PSSOptions{})
+}
+
+// SignArgs contains arguments to a crypto Signer.Sign method.
+type SignArgs struct {
+	Digest []byte
```

Comment:
Can you add comments for each of the field?

Reply:
Done.